common-service/src/main/java/org/sunbird/util/authentication/AuthenticationHelper.java

```diff
@@ -0,0 +1,193 @@
+package org.sunbird.util.authentication;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.cassandra.CassandraOperation;
+import org.sunbird.common.models.response.Response;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.models.util.PropertiesCache;
+import org.sunbird.helper.ServiceFactory;
+import org.sunbird.services.sso.SSOManager;
+import org.sunbird.services.sso.SSOServiceFactory;
+
+/**
+ * This class will handle all the method related to authentication. For example verifying user
+ * access token, creating access token after success login.
+ *
+ * @author Manzarul
+ */
+public class AuthenticationHelper {
+  static {
+    // Application.checkCassandraConnection();
+  }
+
+  private static boolean ssoEnabled =
+      (StringUtils.isNotBlank(System.getenv(JsonKey.SSO_PUBLIC_KEY))
+          && Boolean.parseBoolean(
+              PropertiesCache.getInstance().getProperty(JsonKey.IS_SSO_ENABLED)));
+  private static CassandraOperation cassandraOperation = ServiceFactory.getInstance();
+  private static final String USER_AUTH_TABLE_NAME = "user_auth";
+  private static final String MASTER_KEY_TABLE_NAME = "client_info";
+  private static final String USER_TABLE_NAME = "user";
+  private static final String KEYSPACE_NAME = "sunbird";
+  private static final String USER_EXTERNAL_IDENTITY_TABLE_NAME = "user_external_identity";
+
+  /**
+   * This method will verify the incoming user access token against store data base /cache. If token
```

Comment:
Authenticates given user JWT token.

token User JWT token

User for successfully authenticated request. For unauthenticated requests, UNAUTHORIZED is returned.